api-metastore/src/main/java/org/zalando/nakadi/controller/SchemaController.java

```diff
@@ -53,6 +58,49 @@ public SchemaController(final SchemaService schemaService, final EventTypeServic
         return status(HttpStatus.OK).build();
     }
 
+    @RequestMapping(value = "/event-types/{name}/schemas/{version}/compatibility-check", method = RequestMethod.POST)
+    public ResponseEntity<?> checkCompatibility(@PathVariable("name") final String name,
+                                                @PathVariable("version") final String version,
+                                                @Valid @RequestBody final EventTypeSchemaBase schema,
+                                                final Errors errors) {
+        if (errors.hasErrors()) {
+            throw new ValidationException(errors);
+        }
+
+        final EventType eventType = eventTypeService.get(name);
+
+        if(!schema.getType().equals(eventType.getSchema().getType())){
```

Comment:
Yes, because the check for incompatibilities schemaevolution service throws SchemaEvolutionException and that is mapped to 422 status code already. I wanted to respond with 409 (CONFLICT) particularly when using the compatibility api.

Reply:
Well, for that purpose the common approach is to use different kind of exception and map it in controller exception mapper. Thus the logic will be stored at the same place, and controller will be clean.

Reply:
As discussed removed 409 altogether